File: Day108_2025/Q108_Leetcode.java
class Q108_Leetcode {
    public String countAndSay(int n) {
        if (n == 1) {
            return "1";
        }
        
        String prevSay = countAndSay(n - 1);
        StringBuilder currSay = new StringBuilder();
        int count = 1;
        
        for (int i = 0; i < prevSay.length(); ++i) {
            if (i + 1 < prevSay.length() && prevSay.charAt(i) == prevSay.charAt(i + 1)) {
                count++;
            } else {
                currSay.append(count).append(prevSay.charAt(i));
                count = 1;
            }
        }
        
        return currSay.toString();
    }
}

File: Day108_2025/Q108_Leetcode.py
class Solution:
    def countAndSay(self, n: int) -> str:
        if n == 1:
            return "1"
        
        prev_say = self.countAndSay(n - 1)
        curr_say = ""
        count = 1
        
        for i in range(len(prev_say)):
            if i + 1 < len(prev_say) and prev_say[i] == prev_say[i + 1]:
                count += 1
            else:
                curr_say += str(count) + prev_say[i]
                count = 1
        
        return curr_say
